py_gtfs_rt_ingestion/dev_test_setup.py

```diff
@@ -0,0 +1,298 @@
+    print(
+        f"Starting copy of {num_to_copy} random objects from {src_uri_root} ..."
+    )
+    for uri in random.sample(tuple(uri_copy_set), num_to_copy):
+        key = str(uri).replace(f"s3://{GTFS_BUCKET}/", "")
+        copy_source = {
+            "Bucket": GTFS_BUCKET,
+            "Key": key,
+        }
+        try:
+            s3_client.copy(
+                copy_source,
+                DEV_INGEST_BUCKET,
+                os.path.join(LAMP_PREFIX, key),
+            )
+        except Exception as e:
+            logging.error("Copy failed for: %s", key)
+            logging.exception(e)
+        else:
+            success_count += 1
+    return success_count
+
+
+def drill_s3_folders(
+    client: boto3.client, bucket: str, prefix: str
+) -> Iterable[str]:
+    """
+    Enumerate folders in specified bucket prefix combination
+    """
+    response = client.list_objects_v2(
+        Bucket=bucket, Delimiter="/", Prefix=prefix, MaxKeys=45
+    )
+    if "CommonPrefixes" not in response:
+        yield prefix
+    else:
+        for new_prefix in response["CommonPrefixes"]:
+            yield from drill_s3_folders(client, bucket, new_prefix["Prefix"])
+
+
+def copy_gfts_to_ingest(args: SetupArgs) -> None:
+    """
+    Enumerate folders in GTFS_BUCKET prefix and create workers to copy
+    objects to development Ingest bucket
+    """
+    src_uri_root = f"s3://{os.path.join(GTFS_BUCKET, args.src_prefix)}"
+    dest_urc_root = f"s3://{os.path.join(DEV_INGEST_BUCKET, LAMP_PREFIX)}"
+
+    print(f"Enumerating folders in ({src_uri_root})...")
+    s3_client = boto3.client("s3")
+    folders = list(drill_s3_folders(s3_client, GTFS_BUCKET, args.src_prefix))
+    # Get number of files to pull per folder
+    per_folder = args.objs_to_copy // len(folders)
+    # Number of folders that will have extra file
+    remain = args.objs_to_copy % len(folders)
+    print(
+        f"{len(folders):,} folders found, "
+        f"will copy ~{per_folder} objects from each folder."
+    )
+
+    # Combine list of folders and number of files to pull from each folder
+    folder_pull_cnt = tuple(
+        zip(
+            folders,
+            [
+                per_folder + 1 if x < remain else per_folder
+                for x in range(len(folders))
+            ],
+        )
+    )
+
+    action = None
+    while action not in ("n", "no"):
+        if action in ("y", "yes"):
+            pool_size = os.cpu_count()
+            if pool_size is None:
+                pool_size = 4
+            else:
+                pool_size *= 2
+            with Pool(pool_size) as pool:
+                results = pool.starmap_async(copy_obj, folder_pull_cnt)
+                success_count = 0
+                for result in results.get():
+                    success_count += result
+            print(
+                f"{success_count:,} objects copied, "
+                f"{args.objs_to_copy - success_count} errors."
+            )
+            break
```

Comment:
Create object copy workers, will create 2x he number of cpu's found on the machine. These are low computation but high latency operations, so more workers == more better, I think...